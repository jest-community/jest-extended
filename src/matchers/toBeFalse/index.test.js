import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('.toBeFalse', () => {
  it('passes when given false', () => {
    expect(false).toBeFalse();
  });

  it('fails when not given false', () => {
    expect(() => expect(true).toBeFalse()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeFalse', () => {
  each([[true], [''], [0], [{}], [[]], [() => {}], [undefined], [null], [NaN]]).it(
    'passes when not given false: %s',
    given => {
      expect(given).not.toBeFalse();
    }
  );

  it('fails when given false', () => {
    expect(() => expect(false).not.toBeFalse()).toThrowErrorMatchingSnapshot();
  });
});
