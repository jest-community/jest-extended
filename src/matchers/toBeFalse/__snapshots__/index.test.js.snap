// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeFalse fails when given false 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeFalse(</><dim>)</>

Expected value to not be false received:
  <red>false</>"
`;

exports[`.toBeFalse fails when not given false 1`] = `
"<dim>expect(</><red>received</><dim>).toBeFalse(</><dim>)</>

Expected value to be false:
  <green>false</>
Received:
  <red>true</>"
`;
